src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/configs/GenericRcaConfig.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.configs;
+
+import java.util.Map;
+
+/**
+ * base class config object to store rca config settings
+ */
+public abstract class GenericRcaConfig {
```

Comment:
I don't think we need a `GenericConfig` and then its inherited classes. The json structure inside `rca-config-settings` is similar for all types.

Maybe it would be less code, if we just have a a ConfigHelper class that initializes with RcaConf object and then specific RCAs can ask for a value given a string.

Reply:
done. removed the inheritance and add the below function to retrieve value mapped by (rcaName, setting name ) key. 
public <T> T readRcaConfig(String rcaName, String key, Class<? extends T> clazz)